### PR TITLE
ViewService enhancement to control Lifetime of secondary windows

### PR DESCRIPTION
**Pull request by Viachaslau-Zinkevich**
**Perhaps should wait before stable version is released.**

ViewService enhancement to create secondary views and control their Lifetime. Should be used instead of INavigationService.OpenAsync.
Before using should be statically initialized in App constructor through
`
public App()
 {
    InitializeComponent();
    ViewService.Initialize(this);
}`

Added example to Samples/MultipleViews.

### CODE REVIEW

File: Template10 (Library)/Services/NavigationService/INavigationService.cs [hunk 1]
```diff
@@ -23,6 +23,7 @@ public interface INavigationService
 
         void Refresh();
 
+        [Obsolete("Use Template10.Services.ViewService.IViewService.OpenAsync instead")]
```
Comment (JerryNixon) [inline Template10 (Library)/Services/NavigationService/INavigationService.cs]: Let's consider a way to leave this alone. Could we change the signature to return an instance of IViewService or something to encapsulate the use of the service more simply for developers using T10?

**Comment by Viachaslau-Zinkevich** [comment]
Hm, makes sense. Would you prefer using IViewService inside NavigationService and changed NavigationService.OpenAsync signature to return ViewLifeController?
Also there should be initialization to subscribe for OnWindowCreated. Currently I left this to user to call in constructor, but I would have it done somewhere automatically. Any suggestions?

**Comment by JerryNixon** [comment]
The goal is to keep the current behavior without requiring a developer to learn a new approach, just give the more. We'll need a ViewLife interface.

**Comment by JerryNixon** [comment]
I've change my mind about this I think will go ahead and do it but wait until version 2 when we have redstone supported.

**Comment by JerryNixon** [comment]
I want INavigationService to be the same, for sure. That's used all over the place and in thousands of apps. We can't change that. But we can change the signature of OpenAsync()

https://github.com/Windows-XAML/Template10/blob/master/Template10%20(Library)/Services/NavigationService/NavigationService.cs#L186

Since this is only Task, changing the return signature is not a problem. Right? No change to the developer using it, but developers who understand ViewService can get extra capabilities.

**Comment by JerryNixon** [comment]
You okay with that change?

**Comment by Viachaslau-Zinkevich** [comment]
@JerryNixon absolutely. I can utilize ViewService inside NavigationService as delegating call. Would it be okay for you if I add ViewService initialization(the part that subscribes to OnWindowCreated event) somewhere inside, so that users won't have to bother about this thing?
I swear I got a message about a comment where you were talking about Redstone and v2..but don't see it here

**Comment by JerryNixon** [comment]
The more "taken care of" the better. If you can encapsulate all of the work and the developer just gets benefit for free, then we should definitely do it. 

The Redstone message was actually for another thread and I posted it here by mistake. I think I said I would wait until RS1, but that is NOT what I want to do with this. If you get the PR in before Friday, then I would like to include it in the next NuGet release.

**Comment by Viachaslau-Zinkevich** [comment]
@JerryNixon I updated NavigationService to utilize ViewService inside and added some initialization into BootStrapper

**Comment by JerryNixon** [comment]
Very nice. Needs an update to merge, unfortunately.

**Comment by Viachaslau-Zinkevich** [comment]
Updated, hope to see it without conflicts now

**Comment by Viachaslau-Zinkevich** [comment]
@JerryNixon Guess,can be merged now

**Comment by Viachaslau-Zinkevich** [comment]
@JerryNixon would you please notify here, when you will have plans to merge so that I resolved conflicts then?

**Comment by JerryNixon** [comment]
We're in the process of marking all the services as Sealed. Do that, please, and go ahead and sync. The code looks good. Send me an email jnixon@microsoft when you are ready so I can jump on it and you don't have to sync up again.

**Comment by Viachaslau-Zinkevich** [comment]
Hi @JerryNixon , Sorry for delay, I've updated service with sealed modifier and merged

**Pull request merged**